src/views/Checkout.vue

```diff
@@ -1,43 +1,203 @@
+
+            // Reduce userfriendly addresses from that
+            const addresses = accounts.map((account) => account.userFriendlyAddress);
+
+            // Get balances through pico consensus, also triggers head-change event
+            const network = (this.$refs.network as Network);
+            const balances: Map<string, number> = await network.connectPico(addresses);
+
+            // Update accounts with their balances
+            // (The accounts are still references to themselves in the wallets' accounts maps)
+            for (const account of accounts) {
+                const balance = balances.get(account.userFriendlyAddress);
+                if (balance === undefined) continue;
+                account.balance = Nimiq.Policy.coinsToSatoshis(balance);
+            }
+
+            // Store updated wallets
+            for (const wallet of wallets) {
+                // Update IndexedDB
+                await WalletStore.Instance.put(wallet);
+
+                // Update Vuex
+                this.$addWallet(wallet);
+            }
+
+            // Cache height (balances are stored in IndexedDB)
+            const cacheInput = {
+                timestamp: Date.now(),
+                height: this.height,
+            };
+            window.sessionStorage.setItem(Checkout.BALANCE_CHECK_STORAGE_KEY, JSON.stringify(cacheInput));
+        }
+
+        // Remove loader
+        this.hasBalances = true;
+    }
+
+    private onHeadChange(head: Nimiq.BlockHeader | {height: number}) {
+        this.height = head.height;
+    }
+
+    private accountSelected(walletId: string, address: string) {
+        if (!this.height) return; // TODO: Make it impossible for users to click when height is not ready
+
+        const walletInfo = this.wallets.find((wallet: WalletInfo) => wallet.id === walletId)!;
+        const accountInfo = walletInfo.accounts.get(address)!;
+
+        // FIXME: Currently unused, but should be reactivated
+        this.$store.commit('setActiveAccount', {
+            walletId: walletInfo.id,
+            userFriendlyAddress: accountInfo.userFriendlyAddress,
+        });
+
+        this.proceedToKeyguard(walletInfo, accountInfo);
+    }
+
+    private proceedToKeyguard(walletInfo: WalletInfo, accountInfo: AccountInfo) {
+        // The next block is the earliest for which tx are accepted by standard miners
+        const validityStartHeight = this.height + 1
+            - TX_VALIDITY_WINDOW
+            + this.request.validityDuration;
+
+        const request: KeyguardRequest.SignTransactionRequest = {
+            layout: 'checkout',
+            shopOrigin: this.rpcState.origin,
+            appName: this.request.appName,
+            shopLogoUrl: this.request.shopLogoUrl,
+
+            keyId: walletInfo.id,
+            keyPath: accountInfo.path,
+            keyLabel: walletInfo.label,
+
+            sender: accountInfo.address.serialize(),
+            senderType: Nimiq.Account.Type.BASIC,
```

Comment:
So this is currently not compatible with vesting contracts, right?

Reply:
No, because vesting contracts are not supported currently. They are not displayed for selection and not used anywhere yet.

Reply:
Implementation and support for contracts is tracked in #124.